adrian-greksa/wk01/4-thurs/creditCardValidation/js/creditCardValidation.js

```diff
@@ -0,0 +1,76 @@
+// # Credit Card Validation
+
+// You're starting your own credit card business. You've come up with a new way to validate credit cards with a simple function called validateCreditCard that returns true or false.
+
+// Here are the rules for a valid number:
+
+// - Number must be 16 digits, all of them must be numbers
+// - You must have at least two different digits represented (all of the digits cannot be the same)
+// - The final digit must be even
+// - The sum of all the digits must be greater than 16
+
+// The following credit card numbers are valid:
+
+// - `9999-9999-8888-0000`
+// - `6666-6666-6666-1666`
+
+// The following credit card numbers are invalid:
+
+// - `a923-3211-9c01-1112` invalid characters
+// - `4444-4444-4444-4444` only one type of number
+// - `1111-1111-1111-1110` sum less than 16
+// - `6666-6666-6666-6661` odd final number
+
+// ## Example
+// ```
+// validateCreditCard('9999-9999-8888-0000'); // Returns: true
+// ```
+
+
+const validateCreditCard = function (creditCardNumber) {
+  let cardCharacters = creditCardNumber.replaceAll('-', ''); // .replaceAll(/-/g, '')
+  console.log(cardCharacters);
+
+  if (cardCharacters.length === 16 && cardCharacters.isInteger) { //check if 16 & are numbers
+    if (checkUnique()) {  //check if at least 2 unique characters
+      if (cardCharacters[15] % 2 === 0) {
+        if ()
+      } 
+    }
+  }    
+}
+
+const checkUnique = function() {
+  for (i = 0, i < 16; i++) {
+    let x = true;
+    if (cardCharacters[0] === cardCharacters[i]) {
+      x = false
+    } else {
+      x = false;
+    }
+  }
+}
+*/
+
+const checkSum = function() {
+  let sum = '';
```

Comment:
```suggestion
  let sum = 0;
```
Simple mistake here. If you set sum to a string then the `+=` operator on _line 58_ will **concatenate** instead of **adding**.